roles/openshift_on_openstack_scaleup/tasks/main.yml

```diff
@@ -1,71 +1,122 @@
+# The current node count is the starting node number.
+- name: Setting the start number
   set_fact:
-    inventory_py: "{{ openshift_openstack_dir }}/inventory.py"
-  
-# Get some debug information that makes it easier to file bugs.
-- name: Running commands that create debug information needed to open issues
-  shell: "{{ item }}"
-  register: debug_information
-  with_items:
-    - "printf \"$(uname -srm)\n$(cat /etc/os-release)\n\""
-    - "ansible --version"
-    # - "pip freeze | grep -i jinja"
-    # - "pip freeze | grep -i shade"
-    - "cd openshift-ansible ; git describe"
-    - "rpm -q atomic-openshift-utils openshift-ansible"
-  changed_when: false
-  ignore_errors: true
+    start_node_number: "{{ current_node_count }}"
 
-# Print the debug information out.
-- name: Printing the debug information to the screen
-  debug:
-    msg: "{{ item['stdout_lines'] }}"
-  with_items: "{{ debug_information['results'] }}"
+# The current node count + difference - 1 is the ending node number.
+- name: Setting the end number
+  set_fact:
+    end_node_number: "{{ current_node_count | int + difference | int - 1 }}"
 
-# Set the path for the OpenStack scaleup log file.
-- name: Creating the OpenStack scaleup log file variable
+# Create the new nodes inventory file name
+- name: Creating the file name for the new nodes inventory
   set_fact:
-    openstack_scaleup_log: "{{ ansible_user_dir }}/openstack_scaleup.log"
+    new_nodes_inventory: "{{ ansible_user_dir }}/new_nodes_{{ start_node_number }}-{{ end_node_number }}_inventory"
+
+# Search for regular expressions and replace in file.
+- name: Replacing specific lines in {{ all_yml }}
+  lineinfile:
+    path: {{ all_yml }}
+    regexp: "{{ item['find'] }}"
+    line: "{{ item['replace'] }}"
+  with_items:
+    - { find: "^#?openshift_openstack_num_nodes.*", replace: "openshift_openstack_num_nodes: {{ openshift_node_target }}" }
 
 # Run the Ansible playbook that creates the OpenStack resources.
 - name: Creating the OpenStack resources
-  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/provision.yml -e openshift_repos_enable_testing=true 2>&1 > {{ openstack_scaleup_log }}"
+#  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_cluster_directory }}/provision.yml -e openshift_repos_enable_testing=true 2>&1 > {{ openstack_scaleup_log }}"
+  shell: >
+    {{ ansible_playbook }} -vvv
+    --user openshift
+    -i inventory
+    -i {{ inventory_py }}
+    {{ openshift_cluster_directory }}/provision.yml 2>&1 >> {{ openstack_scaleup_log }}
+  args:
+    # Use bash to get the posix style redirects.
+    executable: /bin/bash
+
+# Get the new OpenStack VMs by name.
+- name: Getting the new OpenStack VM network information by name
+  shell: "{{ openstack }} server list --name app-node-{{ item }}.scale-ci.example.com --format value -c Networks"
```

Comment:
Can the domain be made into a variable?

Reply:
Yes good comment. We could construct the string from variables in all.yml but this code does not access all.yml at this time. I will have to think about this more.

Reply:
Created variables for these values in PR #37